### PR TITLE
Hyperv : Fixed test cases escaped outputs 

### PR DESCRIPTION
**Pull request by asofron**
## Description
Partial fix for #763
Fixed Live Migration escaped outputs
Fixed Backup test cases failure and fix escaped  outputs
Fixed Secureboot escaped outputs
Issue # 
Some of the functions in test cases escape output and make the results of the test case contains those output this need to be fix so the framework doesn`t encounter issues when getting test case result 

Test Result #
 ```
[LISAv2 Test Results Summary]
Test Run On           : 02/24/2019 17:16:53
VHD Under Test        : ubuntu_18.04.1.vhdx
Test Category         : Functional
Test Area             : BACKUP
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:18
[hyperv]
   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 STOR-VSS-BACKUPRESTORE-MULTIFS-VHDX                                               PASS                16.94 
[hyperv]
[hyperv]


[LISAv2 Test Results Summary]
Test Run On           : 02/24/2019 17:36:22
VHD Under Test        : ubuntu_18.04.1.vhdx
Test Category         : Functional
Test Area             : BACKUP
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:22

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 STOR-VSS-BACKUPRESTORE-ISO-NONETWORK                                              PASS                20.72 



[LISAv2 Test Results Summary]
Test Run On           : 02/24/2019 18:01:49
VHD Under Test        : ubuntu_18.04.1.vhdx
Test Category         : Functional
Test Area             : BACKUP
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:16

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 STOR-VSS-BACKUP-PRESTORE-STATE-PAUSED                                             PASS                14.64 


[LISAv2 Test Results Summary]
Test Run On           : 02/24/2019 18:26:34
VHD Under Test        : ubuntu_18.04.1.vhdx
Test Category         : Functional
Test Area             : BACKUP
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:17

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 STOR-VSS-BACKUP-RESTORE-MOUNT-MULTI-PATHS                                         PASS                15.84 


[LISAv2 Test Results Summary]
Test Run On           : 02/24/2019 19:03:23
VHD Under Test        : ubuntu_18.04.1.vhdx
Test Category         : Functional
Test Area             : BACKUP
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:17

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 STOR-VSS-BACKUP-RESTORE-MOUNT-SQUASHFS                                            PASS                15.84 




[LISAv2 Test Results Summary]
Test Run On           : 02/24/2019 20:42:28
VHD Under Test        : Centos_7.5_x64.vhdx
Test Category         : Functional
Test Area             : BACKUP
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:27

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 STOR-VSS-BACKUP-CHANGE-HYPERVVSSD                                                 PASS                26.54 


Note : Only support for RHEL based distros


[LISAv2 Test Results Summary]
Test Run On           : 02/24/2019 21:18:02
VHD Under Test        : ubuntu_18.04.1.vhdx
Test Category         : Functional
Test Area             : BACKUP
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:18

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 STOR-VSS-BACKUP-RESTORE-DISKSTRESS                                                PASS                17.04 




[LISAv2 Test Results Summary]
Test Run On           : 02/24/2019 21:57:05
VHD Under Test        : Centos_7.5_x64.vhdx
Test Category         : Functional
Test Area             : BACKUP
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:29

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 STOR-VSS-BACKUP-DISABLE-ENABLE-VSS                                                PASS                27.91 

Note : Only support for RHEL based distros

[LISAv2 Test Results Summary]
Test Run On           : 02/24/2019 22:28:50
VHD Under Test        : ubuntu_18.04.1.vhdx
Test Category         : Functional
Test Area             : BACKUP
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:17

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 STOR-VSS-BACKUP-RESTORE-FAIL                                                      PASS                15.46 






[LISAv2 Test Results Summary]
Test Run On           : 02/25/2019 18:42:53
VHD Under Test        : ubuntu_18.04.1.vhdx
Test Category         : Functional
Test Area             : MIGRATION
Total Test Cases      : 10 (10 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:58

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 LIVE-MIGRATE                                                                      PASS                 0.51
    2 LIVE-MIGRATE-FAILOVER                                                             PASS                 1.45
    3 LIVE-MIGRATE-MEM4GB-CORE8                                                         PASS                 2.87
    4 LIVE-MIGRATE-MEM8GB-CORE16                                                        PASS                 3.36
    5 LIVE-MIGRATE-MEM16GB-CORE32                                                       PASS                   10
    6 LIVE-MIGRATE-QUICK-MIGRATE                                                        PASS                 1.49
    7 LIVE-MIGRATE-COPYFILE-DURING-MIGRATION                                            PASS                 1.58
    8 LIVE-MIGRATE-QUICKMIG-MEM4GB-CORE8                                                PASS                 2.96
    9 LIVE-MIGRATE-QUICKMIG-MEM8GB-CORE16                                               PASS                 3.37
   10 LIVE-MIGRATE-QUICKMIG-MEM16GB-CORE32                                              PASS                 4.52



[LISAv2 Test Results Summary]
Test Run On           : 02/25/2019 20:15:09
VHD Under Test        : ubuntu_18.04_x64_gen2vm_lisav2.vhdx
Test Category         : Functional
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:2

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 SECUREBOOT-BASIC                                                                  PASS                 1.29 

[LISAv2 Test Results Summary]
Test Run On           : 02/26/2019 06:08:01
VHD Under Test        : ubuntu_18.04_x64_gen2vm_lisav2.vhdx
Test Category         : Functional
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:17

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 SECUREBOOT-VSS                                                                    PASS                16.45 

```

### CODE REVIEW

File: Libraries/IntegrationServiceLibrary.psm1 [hunk 1]
```diff
@@ -905,7 +905,7 @@ Function Check-VMStateAndFileStatus {
 	# only check restore file when IP available
 	$stsipv4 = Test-NetConnection $VMIpv4 -Port 22 -WarningAction SilentlyContinue
 	if ($stsipv4.TcpTestSucceeded) {
-		$sts=Check-FileInLinuxGuest -VMPassword $password -VMPort $VMPort -VMUserName $user -Ipv4 $VMIpv4 -fileName "/home/$user/1"
+		$sts = Run-LinuxCmd -username $user -password $password -ip $VMIpv4 -port $VMPort -command " stat /home/$user/1"
```
Comment (ader1990) [inline Libraries/IntegrationServiceLibrary.psm1]: extra space in the command

File: Testscripts/Windows/SET-SECUREBOOT.ps1 [hunk 2]
```diff
@@ -40,8 +40,8 @@ function Main {
             Write-LogInfo "Successfully Enabled SecureBoot"
         }
         Write-LogInfo "Setting SecureBoot template"
-        Set-VMFirmware -VMName $VMName -ComputerName $HvServer -SecureBootTemplate MicrosoftUEFICertificateAuthority
-        if (-not $?) {
+        $secureboot = Set-VMFirmware -VMName $VMName -ComputerName $HvServer -SecureBootTemplate MicrosoftUEFICertificateAuthority
```
Comment (ader1990) [inline Testscripts/Windows/SET-SECUREBOOT.ps1]: this is not safe at all. explain what the output is and why you are using the output.

File: Testscripts/Windows/SETUP-Backup-DISK.ps1 [hunk 3]
```diff
@@ -44,7 +46,9 @@ function Main {
         if ($currentRetryCount -eq $maxRetryCount) {
             throw "Mounting VHD Failed"
         }
-        $backupdisk = Get-Vhd -Path $backupdiskpath
+        Start-Sleep -Seconds 30
```
Comment (ader1990) [inline Testscripts/Windows/SETUP-Backup-DISK.ps1]: why do you sleep before the mount?

File: Testscripts/Windows/STOR-VSS-BackupRestore-ISO-NoNetwork.ps1 [hunk 4]
```diff
@@ -40,7 +40,7 @@ function Main {
         # Change the working directory to where we need to be
         Set-Location $WorkingDirectory
         $sts = New-BackupSetup $VMName $HvServer
-        if (-not $sts[-1]) {
```
Comment (ader1990) [inline Testscripts/Windows/STOR-VSS-BackupRestore-ISO-NoNetwork.ps1]: why do you change this value -- please give an example output

File: Libraries/IntegrationServiceLibrary.psm1 (hunk 1 above)
Comment (asofron) [reply Libraries/IntegrationServiceLibrary.psm1]: Fixed

File: Testscripts/Windows/SET-SECUREBOOT.ps1 (hunk 2 above)
Comment (asofron) [reply Testscripts/Windows/SET-SECUREBOOT.ps1]: Not using it .Fixed

File: Testscripts/Windows/SETUP-Backup-DISK.ps1 (hunk 3 above)
Comment (asofron) [reply Testscripts/Windows/SETUP-Backup-DISK.ps1]: Removed

File: Testscripts/Windows/STOR-VSS-BackupRestore-ISO-NoNetwork.ps1 (hunk 4 above)
Comment (asofron) [reply Testscripts/Windows/STOR-VSS-BackupRestore-ISO-NoNetwork.ps1]: STS: Enumerating backups... True
 STS -1: True
Reverted change

**Pull request merged**